lib/fedora_migrate/datastream_verification.rb

```diff
@@ -4,28 +4,20 @@ module FedoraMigrate::DatastreamVerification
 
   def valid? datastream=nil
     @datastream = datastream || @source
-    check = has_no_checksum? || has_matching_checksums? || content_equivalent?
+    check = has_matching_checksums? || has_matching_nokogiri_checksums?
     FedoraMigrate::Logger.warn "#{@datastream.pid} datastream #{@datastream.dsid} validation failed" unless check
     check
   end
 
   def has_matching_checksums?
-    datastream.checksum == target_checksum
+    datastream.checksum == target_checksum || Digest::SHA1.hexdigest(datastream.content) == target_checksum
```

Comment:
When the checksum from Fedora3 is "NONE" or if the checksum actually did fail. So we calculate a new one based on the original content from Fedora3.